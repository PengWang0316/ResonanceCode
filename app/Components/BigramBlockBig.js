import React from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/BigramBlockBig.module.css';
import taijiBig from '../imgs/taijiBig.png';
import growthImage from '../imgs/growth.png';
import conceptionImage from '../imgs/conception.png';
import seedingImage from '../imgs/seeding.png';
import maturationImage from '../imgs/maturation.png';

const BigramBlockBig = ({ lineText, position, isLine13 }) => (
  <div className={styles.mainDiv}>
    <div className={styles.lineText}>{lineText}</div>
    <div className={styles.tbTitle}>Agency/Process - Becoming</div>
    <div className="d-flex">
      <div className={`${styles.bigramBlock} ${styles.blockBottomBorder}  ${styles.blockRightBorder} d-flex justify-content-center align-items-center`}>
        <div className={`${position === 1 ? '' : styles.unactive}`}>
          <div className={styles.bigramName}>{isLine13 ? 'Doing/\nProductive' : 'Organizing/\nExpanding'}</div>
          <div className="text-center">
            <img src={growthImage} alt="bigram" />
          </div>
        </div>
      </div>
      <div className={`${styles.bigramBlock} ${styles.blockBottomBorder} d-flex justify-content-center align-items-center`}>
        <div className={`${position === 0 ? '' : styles.unactive}`}>
          <div className={styles.bigramName}>{isLine13 ? 'Transcending/\nEmptying' : 'System Reset'}</div>
          <div className="text-center">
            <img src={conceptionImage} alt="bigram" />
          </div>
        </div>
      </div>
    </div>
    <div className="d-flex">
      <div className={`${styles.bigramBlock} ${styles.blockRightBorder} d-flex justify-content-center align-items-center`}>
        <div className={`${position === 2 ? '' : styles.unactive}`}>
          <div className={styles.bigramName}>{isLine13 ? 'Embodying/\nAttuning' : 'Accruing\nInterest'}</div>
          <div className="text-center">
            <img src={maturationImage} alt="bigram" />
          </div>
        </div>
      </div>
      <div className={`${styles.bigramBlock} d-flex justify-content-center align-items-center`}>
        <div className={`${position === 3 ? '' : styles.unactive}`}>
          <div className={styles.bigramName}>{isLine13 ? 'Composting/\nDigesting' : 'Dispensing/\nConsolidating'}</div>
          <div className="text-center">
            <img src={seedingImage} alt="bigram" />
          </div>
        </div>
      </div>
    </div>
    <div className={styles.tbTitle}>Agency/Process - Being</div>
    <img className={styles.taijiPic} src={taijiBig} alt="taiji" />
    <div className={styles.leftTitle}>Agent/Identity<br />- Being</div>
    <div className={styles.rightTitle}>Agent/Identity<br />- Becoming</div>
  </div>
);
BigramBlockBig.propTypes = {
  lineText: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  isLine13: PropTypes.bool
};
BigramBlockBig.defaultProps = { isLine13: false };
export default BigramBlockBig;
