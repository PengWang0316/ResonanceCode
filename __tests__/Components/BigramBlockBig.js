import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import BigramBlockBig from '../../app/Components/BigramBlockBig';

describe('BigramBlockBig test', () => {
  const defaultProps = {
    lineText: 'test text',
    position: 0,
    isLine13: true
  };
  // const getShallowComponent = (props = defaultProps) => shallow(<BigramBlockBig {...props} />);

  test('BigramBlockBig 0 snapshot', () => expect(renderer.create(<BigramBlockBig {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('BigramBlockBig 1 snapshot', () => expect(renderer.create(<BigramBlockBig {...{
  lineText: 'test text',
  position: 1,
  isLine13: false
}}
  />).toJSON()).toMatchSnapshot());

  test('BigramBlockBig 2 snapshot', () => expect(renderer.create(<BigramBlockBig {...{
  lineText: 'test text',
  position: 2,
  isLine13: false
}}
  />).toJSON()).toMatchSnapshot());

  test('BigramBlockBig 3 snapshot', () => expect(renderer.create(<BigramBlockBig {...{
  lineText: 'test text',
  position: 3,
  isLine13: false
}}
  />).toJSON()).toMatchSnapshot());
});
