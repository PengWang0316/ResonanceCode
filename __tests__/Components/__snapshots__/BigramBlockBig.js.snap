// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigramBlockBig test BigramBlockBig 0 snapshot 1`] = `
<div
  className="mainDiv"
>
  <div
    className="lineText"
  >
    test text
  </div>
  <div
    className="tbTitle"
  >
    Agency/Process - Becoming
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockBottomBorder  blockRightBorder d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          Doing/
Productive
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="bigramBlock blockBottomBorder d-flex justify-content-center align-items-center"
    >
      <div
        className=""
      >
        <div
          className="bigramName"
        >
          Transcending/
Emptying
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockRightBorder d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          Embodying/
Attuning
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="bigramBlock d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          Composting/
Digesting
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="tbTitle"
  >
    Agency/Process - Being
  </div>
  <img
    alt="taiji"
    className="taijiPic"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="leftTitle"
  >
        Agent
    <br />
    Perspective
    <br />
        Being
  </div>
  <div
    className="rightTitle"
  >
        Agent
    <br />
    Perspective
    <br />
      Becoming
  </div>
</div>
`;

exports[`BigramBlockBig test BigramBlockBig 1 snapshot 1`] = `
<div
  className="mainDiv"
>
  <div
    className="lineText"
  >
    test text
  </div>
  <div
    className="tbTitle"
  >
    Agency/Process - Becoming
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockBottomBorder  blockRightBorder d-flex justify-content-center align-items-center"
    >
      <div
        className=""
      >
        <div
          className="bigramName"
        >
          Organizing/
Expanding
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="bigramBlock blockBottomBorder d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          System Reset
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockRightBorder d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          Accruing
Interest
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="bigramBlock d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          Dispensing/
Consolidating
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="tbTitle"
  >
    Agency/Process - Being
  </div>
  <img
    alt="taiji"
    className="taijiPic"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="leftTitle"
  >
        Agent
    <br />
    Perspective
    <br />
        Being
  </div>
  <div
    className="rightTitle"
  >
        Agent
    <br />
    Perspective
    <br />
      Becoming
  </div>
</div>
`;

exports[`BigramBlockBig test BigramBlockBig 2 snapshot 1`] = `
<div
  className="mainDiv"
>
  <div
    className="lineText"
  >
    test text
  </div>
  <div
    className="tbTitle"
  >
    Agency/Process - Becoming
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockBottomBorder  blockRightBorder d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          Organizing/
Expanding
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="bigramBlock blockBottomBorder d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          System Reset
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockRightBorder d-flex justify-content-center align-items-center"
    >
      <div
        className=""
      >
        <div
          className="bigramName"
        >
          Accruing
Interest
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="bigramBlock d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          Dispensing/
Consolidating
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="tbTitle"
  >
    Agency/Process - Being
  </div>
  <img
    alt="taiji"
    className="taijiPic"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="leftTitle"
  >
        Agent
    <br />
    Perspective
    <br />
        Being
  </div>
  <div
    className="rightTitle"
  >
        Agent
    <br />
    Perspective
    <br />
      Becoming
  </div>
</div>
`;

exports[`BigramBlockBig test BigramBlockBig 3 snapshot 1`] = `
<div
  className="mainDiv"
>
  <div
    className="lineText"
  >
    test text
  </div>
  <div
    className="tbTitle"
  >
    Agency/Process - Becoming
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockBottomBorder  blockRightBorder d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          Organizing/
Expanding
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="bigramBlock blockBottomBorder d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          System Reset
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="d-flex"
  >
    <div
      className="bigramBlock blockRightBorder d-flex justify-content-center align-items-center"
    >
      <div
        className="unactive"
      >
        <div
          className="bigramName"
        >
          Accruing
Interest
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="bigramBlock d-flex justify-content-center align-items-center"
    >
      <div
        className=""
      >
        <div
          className="bigramName"
        >
          Dispensing/
Consolidating
        </div>
        <div
          className="text-center"
        >
          <img
            alt="bigram"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="tbTitle"
  >
    Agency/Process - Being
  </div>
  <img
    alt="taiji"
    className="taijiPic"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="leftTitle"
  >
        Agent
    <br />
    Perspective
    <br />
        Being
  </div>
  <div
    className="rightTitle"
  >
        Agent
    <br />
    Perspective
    <br />
      Becoming
  </div>
</div>
`;
